### ISSUE TITLE
Strange Local Variable Behaviour

### ISSUE THREAD

**Issue by Mummified316** [Pulover/PuloversMacroCreator#133]
If a first function defines a variable and than executes a second function which uses a variable of the same name, the second function will replicate the first function's variable. Furthermore, if the second function, before attempting to use the variable, first defines said variable, than when the second function finishes, the first function's variable will have been reset.

Example of the first function's variable getting used by the second function:
```
TestFunction1()
{
    var1 := "Hey there!"
    MsgBox, 0, , %var1%
    TestFunction2()
}

TestFunction2()
{
    MsgBox, 0, , %var1%
}
```

Example of the first function's variable resetting:
```
TestFunction1()
{
    var1 := "Hey there!"
    MsgBox, 0, , %var1%
    TestFunction2()
    MsgBox, 0, , %var1%
}

TestFunction2()
{
    var1 := "Look out!"
    MsgBox, 0, , %var1%
}
```

**Comment by Pulover** [comment]
Hello,

Thank you for the report and examples. I'm going to investigate and try to fix it.

**Comment by Pulover** [comment]
Fixed in v5.3.1.
